packages/deployments/contracts/contracts/nomad-xapps/contracts/promise-router/PromiseMessage.sol

```diff
@@ -22,13 +22,13 @@ library PromiseMessage {
 
   // ============ Constants ============
   uint256 private constant IDENTIFIER_LEN = 1;
-  // 1 byte identifier + 32 bytes transferId + 20 bytes callback + 32 bytes length + x bytes data
-  // before: 1 byte identifier + 32 bytes transferId + 20 bytes callback = 53 bytes
-  uint256 private constant LENGTH_CALLDATA_START = 53;
-  uint8 private constant LENGTH_CALLDATA_LEN = 32;
+  // 1 byte identifier + 32 bytes transferId + 20 bytes callback + 1 byte success + 32 bytes length + x bytes data
+  // before: 1 byte identifier + 32 bytes transferId + 20 bytes callback + 1 byte success= 54 bytes
+  uint256 private constant LENGTH_RETURNDATA_START = 54;
+  uint8 private constant LENGTH_RETURNDATA_LEN = 32;
```

Comment:
Why does `PromiseMessage` need this extra 1 byte success? Just curious

Reply:
that is the `success` of the calldata executed on the destination domain

Reply:
`success` field is the boolean field that shows if an external call success or not.